middleware/throttle.go

```diff
@@ -2,23 +2,43 @@ package middleware
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/pressly/chi"
 	"golang.org/x/net/context"
 )
 
-// Throttle is a middleware that limits number of currently
-// processed requests at a time.
-func Throttle(limit int) func(chi.Handler) chi.Handler {
-	if limit <= 0 {
+var (
+	defaultTimeout = time.Second * 60
+)
+
+// Throttle is a middleware that limits number of currently processed requests
+// at a time.
+func Throttle(limit int, backloglimit int, timeout time.Duration) func(chi.Handler) chi.Handler {
+	if limit < 1 {
 		panic("middleware.Throttle expects limit > 0")
 	}
 
+	if backloglimit < limit {
+		panic("middleware.Throttle expects backloglimit > limit")
+	}
```

Comment:
This needs to be explained. Seeing backlog in a throttler I assumed that backlog is the number of requests that you can queue on top of the limit, but if I correctly understand the code starting here https://github.com/pressly/chi/pull/26/files#diff-31daf09af7ba24641f7fce5bb0918d1dR76 backlog is the part of the limit that is queued (which explains why `backlog < limit` has to be enforced)